parcels/particle.py

```diff
@@ -280,22 +283,36 @@ def execute(self, pyfunc=AdvectionRK4, starttime=None, endtime=None, dt=1.,
             starttime = starttime.total_seconds()
         if isinstance(endtime, delta):
             endtime = endtime.total_seconds()
+        if isinstance(runtime, delta):
+            runtime = runtime.total_seconds()
         if isinstance(dt, delta):
             dt = dt.total_seconds()
         if isinstance(output_interval, delta):
             output_interval = output_interval.total_seconds()
-
-        # Check if starttime, endtime and dt are consistent and compute timesteps
+        if isinstance(starttime, datetime):
+            starttime = (starttime - self.time_origin).total_seconds()
+        if isinstance(endtime, datetime):
+            endtime = (endtime - self.time_origin).total_seconds()
+
+        # Check if starttime, endtime, runtime and dt are consistent and compute timesteps
+        if runtime is not None and endtime is not None:
+            sys.exit("ERROR: Only one of (endtime, runtime) can be specified")
```

Comment:
Fixed now to suggested pythonic version of error handling (learning something new every day!)